core/src/test/java/org/apache/iceberg/TestWapWorkflow.java

```diff
@@ -684,4 +685,87 @@ public void testNonWapCherrypick() {
           table.manageSnapshots().cherrypick(firstSnapshotId).commit();
         });
   }
+
+  @Test
+  public void testWithExpiringWapThenCherrypick() {
```

Comment:
I think these tests should be in the suite for snapshot expiration, not the suite for WAP. Can you move them?